src/bsp/src/common/src/circular_buff.c

```diff
@@ -0,0 +1,164 @@
+#include "common/circular_buff.h"
+#include <stdbool.h>
+#include <stddef.h>
+
+// TODO: Add assert instead of NULL checks
+bool CircularBuff_init(CircularBuff* circularBuff, uint8_t* buffer, uint16_t size) {
+    if (circularBuff == NULL || buffer == NULL) {
```

Comment:
The circular buffer does not manage its own memory? What if the memory to which the buffer refers is deallocated or goes out of scope? It would pass the initial null check but afterwards, run into issues. It may not be problematic the way it is used now, but would be subject to defects if changes occurred. I would either use a malloc (I know, dynamic = bad, but not necessarily illegal) or have a static buffer of a defined length. A check would be made on the size to make sure the size required is not larger than this buffer.  This would avoid having to constantly run null checks of every function.

Also, why C instead of C++ for the circular buff? I don't mind either, just wanted to know if there was a specific reason why.

Reply:
Hmm, I guess it doesn't manage the memory locally to have multiple instances of the buffer... If it has to be in C, then I would leave it as is.

Reply:
How can it manage it's own memory? If you do a malloc in the init, you need to have to call a deinit, which comes back to the same things as not managing memory. This allows the user to manage it's own memory and thus be more flexible. The user can easily write a small wrapper for the init to match you description if needed.

It needs to be in C since it's used in the low level generated C files and may not be only in C++ files.

Reply:
True! And makes sense to have it in C then.